daemon/access.go

```diff
@@ -166,7 +166,28 @@ var (
 	requireInterfaceApiAccess = requireInterfaceApiAccessImpl
 )
 
-func requireInterfaceApiAccessImpl(d *Daemon, r *http.Request, ucred *ucrednet, interfaceNames []string) *apiError {
+type interfaceAccessReqs struct {
+	// Interfaces is a list of interfaces, at least one of which must be
+	// connected
+	Interfaces []string
+
+	// Slot when true, the snap must appear on the slot side
+	Slot bool
+	// Plug when true, the snap must appear on the plug side
+	Plug bool
```

Comment:
do we need to support setting both to true because the logic below and the descriptions here don't match in that case